idea/gearexe/frontend/src/app/api/use-read-contract-state.ts

```diff
@@ -0,0 +1,25 @@
+import { useQuery } from '@tanstack/react-query';
+import { isAddress } from 'ethers';
+import { HexString } from 'gearexe';
+
+import { useGearExeApi } from '../providers';
+
+import { useMirrorContract } from './use-mirror-contract';
+
+const useReadContractState = (programId: HexString) => {
```

Comment:
Wouldn't be a good idea to collocate hooks with theirs features as well? `useReadContractState` can only be a part of a program.